Budibase/budibase
# Account Portal Password Reset API tests

## Description

**_This PR was originally here - but was marked as stale and closed - Opening a new PR here for the same changes_**

Adding a test which 'performs password reset flow' - Cloud account only, for now

Test Steps:

Create new account
Request password reset to get code
Change password using code
Login using new password
Validate can’t login using old password
Updated AuthAPI.ts

Functions added for logout, resetPassword, resetPasswordUpdate (required for testing)
resetPassword, resetPasswordUpdate functions required an update to misc.ts (RESET_PASSWORD_CODE + RETURN_RESET_PASSWORD_CODE added)
Updated accounts.cloud.internal.spec.ts and accounts.cloud.spec.ts

The usage of 'generateAccount' within these files has been updated to include hosting (cloud in this case)
There are separate changes made to the account portal to support this testing - These changes can be found https://github.com/Budibase/account-portal/pull/388

Further commit to this PR:

Updating the test 'performs signup and deletion flow' to include testing of the self endpoint

The self endpoint is '/api/auth/self'
A check is included to verify the self response is unauthorized (403 status)
A check is included to verify the self response matches the account (200 status)
